scipy/sparse/_dok.py

```diff
@@ -174,8 +185,8 @@ def _get_sliceXslice(self, row, col):
             j, rj = divmod(int(key[1]) - col_start, col_step)
             if rj != 0 or j < 0 or j >= shape[1]:
                 continue
-            x = dict.__getitem__(self, key)
-            dict.__setitem__(newdok, (i, j), x)
+            x = self._dict[key]
+            newdok[i, j] = x
```

Comment:
In this case we can set `newdok._dict[(i, j)] = x` directly, to avoid routing through the general-purpose indexing logic.